### DISCUSSION TITLE
CivitAI API integration

### DISCUSSION THREAD

**Discussion by joshmedeski** [Comfy-Org/ComfyUI-Manager#214] [Ideas]
I think it would be great to integrate [CivitAI's API](https://github.com/civitai/civitai/wiki/REST-API-Reference) to the "Install Models modal" and allow users to install and update models using the ComfyUI Manager.

What do you think?

**Comment by ltdrdata** [comment]
I'll add it to the to-do list.

**Comment by BuildBackBuehler** [comment]
I like the idea! I'd prefer if it just worked as such:

1. Personally, I've been using this Civit Downloader extension knowing the addt'l images/files would come in handy. 
https://www.ayamaru.com/more If this is a ridiculously popular extension used by the Comfy community, it'd make sense to just integrate that into however CivitAI is integrated. I have it setup to download a .info file which contains whatever information would be needed to fetch updates and whatever else.

I think I started a node to do just that. I'm an amateur at best so it may not be of any use. I'll have to take a look @ it. 

So in my implementation of things, I'd planned on having the files piped into Bilbo's Photo Prompt X. Their GUI is amazing. I was also editing that to build in tabs, have categories outside "photography", whatever the user wants. 

2. On the other side of things, it sounds like you're interested in Comfy Users' providing Models that'll be maintained? I do think a curated list would be a great way of cutting down the massive timesuck that is model hunting. Would be more of a luxury than a need IMO, but I'd potentially love that. 

3. I think another thing to consider is just how convoluted the nav. is already becoming. May need to simplify and/or enhance existing functionalities of ComfyUI-Manager. 

Right now we've got a 7x7x4ish board, a thiccc menu. I do think there'd be a yoooge benefit from making the UI into a collapsing menu nav, categorized into X, Y and Z with all the more niche/detailed options avail hidden in the submenus. 

I could provide a visual representation of what I'm thinking if it helps
